File: 003/003.cpp
#include <iostream>
#include <math.h>

// What is the largest prime factor of the number 600851475143?

using namespace std;

int isPrime (int number) {
    for (int j = 2; j < number / 2; j++)
        if (number % j == 0) {
            return 0;
        }
    return 1;
}

int main () {
    double target = 600851475143;
    for (int i = 2; i <= target / 2; i++) {
        double k = target / i;
        if (ceilf(k) == k) {
            if (isPrime((int)k) == 1) {
                cout << fixed << "The highest prime factorial is " << (int)k << endl;
                break;
            }
        }
    }
}

File: 003/003.rb
#!/usr/local/bin/ruby -w

# What is the largest prime factor of the number 600851475143?

target = 600_851_475_143

def prime?(number)
    (2..number.to_i / 2).each do |j|
        return 0 if (number % j).zero?
    end
    return 1
end

(2..target / 2).each do |i|
    k = target.to_f / i
    if (k % 1).zero?
        if prime?(k) == 1
            puts("The highest prime factorial is #{k.to_i}.")
            break
        end
    end
end
